src/pools/xaveFxPool/fxPool.ts

```diff
@@ -0,0 +1,249 @@
+    isLBP?: boolean | undefined;
+
+    _calcTokensOutGivenExactBptIn(bptAmountIn: BigNumber): BigNumber[] {
+        // Will copy over other implementations, not supporting BPT tokens atm
+        return new Array(this.tokens.length).fill(Zero);
+    }
+
+    _calcBptOutGivenExactTokensIn(amountsIn: BigNumber[]): BigNumber {
+        // Will copy over other implementations, not supporting BPT tokens atm
+        return Zero;
+    }
+
+    parsePoolPairData(tokenIn: string, tokenOut: string): FxPoolPairData {
+        const tokenIndexIn = this.tokens.findIndex(
+            (t) => getAddress(t.address) === getAddress(tokenIn)
+        );
+        if (tokenIndexIn < 0) throw 'Pool does not contain tokenIn';
+        const tI = this.tokens[tokenIndexIn];
+        const balanceIn = tI.balance;
+        const decimalsIn = tI.decimals;
+
+        const tokenIndexOut = this.tokens.findIndex(
+            (t) => getAddress(t.address) === getAddress(tokenOut)
+        );
+
+        if (tokenIndexOut < 0) throw 'Pool does not contain tokenOut';
+        const tO = this.tokens[tokenIndexOut];
+        const balanceOut = tO.balance;
+        const decimalsOut = tO.decimals;
+
+        const poolPairData: FxPoolPairData = {
+            id: this.id,
+            address: this.address,
+            poolType: this.poolType,
+            tokenIn: tokenIn,
+            tokenOut: tokenOut,
+            decimalsIn: Number(decimalsIn),
+            decimalsOut: Number(decimalsOut),
+            balanceIn: parseFixed(balanceIn, decimalsIn),
+            balanceOut: parseFixed(balanceOut, decimalsOut),
+            swapFee: this.swapFee,
+            alpha: this.alpha,
+            beta: this.beta,
+            lambda: this.lambda,
+            delta: this.delta,
+            epsilon: this.epsilon,
+            tokenInRate: parseFixed(this.tokens[tokenIndexIn].fxRate!),
+            tokenOutRate: parseFixed(this.tokens[tokenIndexOut].fxRate!),
+        };
+
+        return poolPairData;
+    }
+
+    // Normalized liquidity is an abstract term that can be thought of the
+    // inverse of the slippage. It is proportional to the token balances in the
+    // pool but also depends on the shape of the invariant curve.
+    // As a standard, we define normalized liquidity in tokenOut
+    getNormalizedLiquidity(poolPairData: FxPoolPairData): OldBigNumber {
+        // This could be refined by using the inverse of the slippage, but
+        // in practice this won't have a big impact in path selection for
+        // multi-hops so not a big priority
+        return bnum(
+            formatFixed(poolPairData.balanceOut, poolPairData.decimalsOut)
+        );
+    }
+
+    getLimitAmountSwap(
+        poolPairData: FxPoolPairData,
+        swapType: SwapTypes
+    ): OldBigNumber {
+        if (swapType === SwapTypes.SwapExactIn) {
+            return bnum(
+                formatFixed(
+                    poolPairData.balanceIn.mul(this.MAX_IN_RATIO).div(ONE),
```

Comment:
Just a check - this is currently using the same limits as a weighted pool, e.g. can only trade a 1/3 of token balance in the pool. Is this correct?

Reply:
we decided to remove the limits. our fxpool logic has an alpha region where it halts when we can no longer swap based on our swap invariant. hope this works!

Reply:
The limit should reflect the max amount that can be swapped for that pool, so e.g. for the WeightedPools the limit is a 1/3 of the token balance because that reflects the smart contract limit. 
At the moment you limit is set to the total balance of the pool. But based off your explanation it sounds like your pools will also have an upper limit based on the alpha region and this is what should be reflected here?

Reply:
we added a quick calculation for the alpha region now @johngrantuk. this includes the upper limit on each swap. hope this works!